### ISSUE TITLE
Add option to only prefer-arrow-callback on single statement functions

### ISSUE THREAD

**Issue by NickHeiner** [eslint/eslint#6665]
I would like to have a lint rule that only requires single statement functions to use the arrow callback style:

``` js
// ok
foo(function() {
    console.log('x');
    return y;
});

// not ok
foo(function() {
    return y;
});

// corrected
foo(() => y);
```

Right now, there is no way to express this with the eslint rules as far as I can tell.

I would add this as an option to `prefer-arrow-callback`. Something like:

```
'prefer-arrow-callback': ['warn', {allowMultiStatementFunctions: true}],
```

I am currently using `eslint@3.0.1` and `Node v.4.2.1`. I'm happy to submit a PR for this if you'd accept it.

**Comment by platinumazure** [comment]
Hi @NickHeiner, thanks for the issue! I agree this seems like a worthwhile enhancement, though I wonder if it could generalize to "require arrow functions if statement count is <= some number"?

I've gone ahead and filled out the [rule change request questionnaire](http://eslint.org/docs/developer-guide/contributing/rule-changes#proposing-a-rule-change) for you, please let me know if I've gotten anything wrong. (And in future, please do consider using this as a guideline for rule change requests, it really helps us out!)

## The version of ESLint you are using
ESLint 3.0.1

## The rule you want to change
prefer-arrow-callback

## The code you want to be flagged as ~~incorrect~~ correct
```js
foo(function () {
    console.log('x');
    return y;
});
```

No change requested for functions of one statement in length:
```js
foo(function() {
    return y;    // Is flagged, should still be flagged
});
```

## What happens when the rule is applied to the code without your change
First example shows an error requiring an arrow callback. Second example correctly shows an error (and no change is desired).

**Comment by NickHeiner** [comment]
Yes, I'll use that form in the future. I thought that if I had all the necessary information, it wouldn't matter precisely which format it was in. Thanks for filling out the form for me this time! Your form is correct.

I like the idea of generalizing it to make the upper bound on the number of statements configurable.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
